dvc/tree/webhdfs.py

```diff
@@ -100,7 +100,7 @@ def open(self, path_info, mode="r", encoding=None, **kwargs):
         with self.hdfs_client.read(
             path_info.path, encoding=encoding
         ) as reader:
-            yield reader.read()
+            yield reader
```

Comment:
Just making sure this is not an accidental change.

Reply:
`reader.read()` will return string/bytes, `open()` should return an IO interface that we can call `.read()` on demand

Reply:
ah, strange that it didn't fail in the tests before. Probably I've missed something. Which test started failing now, btw?

Reply:
> Which test started failing now, btw?

https://github.com/iterative/dvc/blob/1a43bfc29d49050b3e7f83d204d9105f9d615503/tests/func/test_tree.py#L326-L327

Reply:
@isidentical Thank you! :pray: